controllers/common.go

```diff
@@ -324,13 +327,65 @@ func (r *MultiClusterHubReconciler) OverrideImagesFromConfigmap(imageOverrides m
 			} else if manifestImage.ImageTag != "" {
 				imageOverrides[manifestImage.ImageKey] = fmt.Sprintf("%s/%s:%s", manifestImage.ImageRemote, manifestImage.ImageName, manifestImage.ImageTag)
 			}
-
 		}
 	}
 
 	return imageOverrides, nil
 }
 
+// OverrideTemplatesFromConfigmap ...
+func (r *MultiClusterHubReconciler) OverrideTemplatesFromConfigmap(templateOverrides map[string]string,
+	namespace, configmapName string) (map[string]string, error) {
+	r.Log.Info(fmt.Sprintf("Overriding template from configmap: %s/%s", namespace, configmapName))
+
+	configmap := &corev1.ConfigMap{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{
+		Name:      configmapName,
+		Namespace: namespace,
+	}, configmap)
+	if err != nil && errors.IsNotFound(err) {
+		return templateOverrides, err
+	}
+
+	if len(configmap.Data) != 1 {
+		return templateOverrides, fmt.Errorf(fmt.Sprintf("Unexpected number of keys in configmap: %s", configmapName))
+	}
+
+	var config manifest.ManifestTemplate
+	for _, v := range configmap.Data {
+		err = json.Unmarshal([]byte(v), &config)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	for key, value := range config.TemplateOverrides {
+		// Convert value to string if necessary
+		strValue, err := convertToString(value)
+		if err != nil {
+			return nil, fmt.Errorf("error converting value for key %s: %w", key, err)
+		}
+		templateOverrides[key] = strValue
```

Comment:
If I'm following along correctly, in _controller.go, we first read in template overrides from env vars, and then if there is an template-overrides-cm annotation we call this function passing the templateOverides map we have after the load-from-env-var stuff.  Here, in this function we iterate over the keys in the json obj from the config map, and for each key was save a key/value entry into the templateOVerrides map.

This achieve a merging of the two sources,  with the effect that if a given key appears in both sources, the value from the configmap key/value entry wins.  Its a close call either way, but I would prefer to see this implemented such that if  given key appears in both sources, the value from the environment variable wins.    My rationale is that such an approach would let us have some "permanent overrides" that are established by creating a config map and making the MCH point to it, and then you could have some "temporary overrides of those overrides" by setting env vars (via OLM subscription spec.config.env).

If you agree, one way to do that would be to check that the key is not already in the map here, before setting the map entry.